src/overlays/actors/ovl_Obj_Kibako/z_obj_kibako.c

```diff
@@ -20,42 +39,375 @@ const ActorInit Obj_Kibako_InitVars = {
+    GlobalContext* globalCtx = globalCtx2;
+    ObjKibako* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void ObjKibako_AirBreak(ObjKibako* this, GlobalContext* globalCtx) {
+    s16 angle;
+    s32 i;
+    Vec3f* worldPos = &this->actor.world.pos;
+    Vec3f pos;
+    Vec3f velocity;
+
+    for (i = 0, angle = 0; i < 12; i++, angle += 0x4E20) {
+        f32 sn = Math_SinS(angle);
+        f32 cs = Math_CosS(angle);
+        f32 temp_rand;
+        s16 phi_s0;
+
+        pos.x = sn * 16.0f;
+        pos.y = (Rand_ZeroOne() * 5.0f) + 2.0f;
+        pos.z = cs * 16.0f;
+        velocity.x = pos.x * 0.2f;
+        velocity.y = (Rand_ZeroOne() * 6.0f) + 2.0f;
+        velocity.z = pos.z * 0.2f;
+        pos.x += worldPos->x;
+        pos.y += worldPos->y;
+        pos.z += worldPos->z;
+        temp_rand = Rand_ZeroOne();
+        if (temp_rand < 0.1f) {
+            phi_s0 = 0x60;
+        } else if (temp_rand < 0.7f) {
+            phi_s0 = 0x40;
+        } else {
+            phi_s0 = 0x20;
+        }
+
+        EffectSsKakera_Spawn(globalCtx, &pos, &velocity, worldPos, -200, phi_s0, 20, 0, 0,
+                             (Rand_ZeroOne() * 38.0f) + 10.0f, 0, 0, 60, -1,
+                             sObjectBankTable[(this->actor.params >> 0xF) & 1],
+                             sParticleDisplayLists[(this->actor.params >> 0xF) & 1]);
+    }
+
+    func_800BBFB0(globalCtx, worldPos, 40.0f, 3, 0x32, 0x8C, 1);
+    func_800BBFB0(globalCtx, worldPos, 40.0f, 2, 0x14, 0x50, 1);
+}
+
+void ObjKibako_WaterBreak(ObjKibako* this, GlobalContext* globalCtx) {
+    s16 angle;
+    s32 i;
+    Vec3f* worldPos = &this->actor.world.pos;
+    Vec3f pos;
+    Vec3f velocity;
+
+    pos.y = worldPos->y + this->actor.yDistToWater;
+    for (angle = 0, i = 0; i < 5; i++, angle += 0x3333) {
+        pos.x = (Math_SinS(((s32)(Rand_ZeroOne() * 6000.0f)) + angle) * 15.0f) + worldPos->x;
+        pos.z = (Math_CosS(((s32)(Rand_ZeroOne() * 6000.0f)) + angle) * 15.0f) + worldPos->z;
+        EffectSsGSplash_Spawn(globalCtx, &pos, NULL, NULL, 0, 350);
+    }
+    pos.x = worldPos->x;
+    pos.z = worldPos->z;
+    EffectSsGRipple_Spawn(globalCtx, &pos, 200, 600, 0);
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Kibako_0x809261B0/func_80926224.asm")
+    for (i = 0, angle = 0; i < 12; i++, angle += 0x4E20) {
+        f32 sn = Math_SinS(angle);
+        f32 cs = Math_CosS(angle);
+        f32 temp_rand;
+        s16 phi_s0;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Kibako_0x809261B0/func_809262BC.asm")
+        pos.x = sn * 16.0f;
+        pos.y = (Rand_ZeroOne() * 5.0f) + 2.0f;
+        pos.z = cs * 16.0f;
+        velocity.x = pos.x * 0.18f;
+        velocity.y = (Rand_ZeroOne() * 4.0f) + 2.0f;
+        velocity.z = pos.z * 0.18f;
+        pos.x += worldPos->x;
+        pos.y += worldPos->y;
+        pos.z += worldPos->z;
+        temp_rand = Rand_ZeroOne();
```

Comment:
I'd call this `ratio` or something instead personally

Reply:
Is it a ratio? This is what it's called in the OoT repo, so I'm going with consistency over a guess.

Reply:
I'd consider anything returned from rand_zeroone to be that